### PR TITLE
Document motivation for a KDF after key-exchange (#4005)

### PR DESCRIPTION
**Pull request by jlaine**

### CODE REVIEW

**Comment by jlaine** [comment]
I don't understand the codecov failure, I haven't touched any code..

**Comment by reaperhulk** [comment]
I still don't have a better suggestion for this language, but in the meantime the examples should be updated to use a KDF.

File: docs/hazmat/primitives/asymmetric/dh.rst [hunk 1]
```diff
@@ -19,7 +19,9 @@ Exchange Algorithm
 ~~~~~~~~~~~~~~~~~~
 
 For most applications the ``shared_key`` should be passed to a key
-derivation function.
+derivation function in order to produce a key with the expected number of
+random bits, and to destroy mathematical structure in the shared key which
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/dh.rst]: After soliciting several people for ideas on phrasing here how about:

"...should be passed to a key derivation function. This allows mixing of additional information into the key, derivation of multiple keys, and destroys any structure that may be present."

Not really much better I guess ;)

**Comment by jlaine** [comment]
For future reference, the output of exectack looks like:

```
- execstack.check/cryptography/hazmat/bindings/_constant_time.abi3.so
- execstack.check/cryptography/hazmat/bindings/_openssl.abi3.so
- execstack.check/cryptography/hazmat/bindings/_padding.abi3.so
```

If I enable exectack on a .so it looks like:

```
- execstack.check/cryptography/hazmat/bindings/_constant_time.abi3.so
X execstack.check/cryptography/hazmat/bindings/_openssl.abi3.so
- execstack.check/cryptography/hazmat/bindings/_padding.abi3.so
```

=> the test is "if any line starts with X", we have a problem

**Pull request merged**